pandas/core/groupby/indexing.py

```diff
@@ -0,0 +1,184 @@
+from __future__ import annotations
+
+from typing import (
+    Iterable,
+    cast,
+)
+
+import numpy as np
+
+from pandas._typing import (
+    FrameOrSeries,
+    PositionalIndexer,
+)
+from pandas.util._decorators import (
+    cache_readonly,
+    doc,
+)
+
+from pandas.core.dtypes.common import (
+    is_integer,
+    is_list_like,
+)
+
+from pandas.core.groupby import groupby
```

Comment:
how are you importing this? (when this file itself is imported by group)

Reply:
yah this looks like a circular import that is going to be really fragile

Reply:
I only did this as a mixin because I was copying the style of df.iloc. Which does not attempt to type itself.
In fact, I do need access to  GroupBy attributes, so I had to use a cast to get it past the tests.
The simple solution would be to abandon trying to statically type this.

Reply:
ok this is *just* for typing, so if you put it in a TYPE_CHECKING block will be ok (e.g .as the groupby.GroupBy should be separated out to avoid deps)

Reply:
I will wait until my `group_selection_context` change is tested and reviewed before sorting out this import and the type checking.

Reply:
It tested OK, so I have added the conditional type checking.;